qiskit_machine_learning/neural_networks/effective_dimension.py

```diff
@@ -0,0 +1,354 @@
+    """
+
+    def __init__(
+        self,
+        qnn: NeuralNetwork,
+        weight_samples: Union[List[float], np.ndarray, int] = 1,
+        input_samples: Union[List[float], np.ndarray, int] = 1,
+        callback: Optional[Callable[[int, float, float], None]] = None,
+    ) -> None:
+
+        """
+        Args:
+            qnn: A Qiskit :class:`~qiskit_machine_learning.neural_networks.NeuralNetwork`,
+                with a specific dimension ``(num_weights)`` that will determine the shape of the
+                Fisher Information Matrix ``(num_input_samples * num_weight_samples, num_weights,
+                num_weights)`` used to compute the global effective dimension for a set of
+                ``input_samples``, of shape ``(num_input_samples, qnn_input_size)``, and
+                ``weight_samples``, of shape ``(num_weight_samples, num_weights)``.
+            weight_samples: An array of neural network parameters (weights), of shape
+                ``(num_weight_samples, num_weights)``, or an ``int`` to indicate the number of
+                parameter sets to sample randomly from a uniform distribution. By default,
+                ``weight_samples = 1``.
+            input_samples: An array of samples to the neural network, of shape
+                ``(num_input_samples, qnn_input_size)``, or an ``int`` to indicate the number of
+                input sets to sample randomly from a normal distribution. By default,
+                ``input_samples = 1``.
+            callback: A callback function for the Monte Carlo sampling.
+        """
+
+        # Store arguments
+        self._weight_samples = None
+        self._input_samples = None
+        self._num_weight_samples = 1
+        self._num_input_samples = 1
+        self._model = qnn
+        self._callback = callback
+
+        # Define weight samples and input samples
+        self.weight_samples = weight_samples  # type: ignore
+        # input setter uses self._model
+        self.input_samples = input_samples  # type: ignore
+
+    def num_weights(self) -> int:
+        """Returns the dimension of the model according to the definition from [1]."""
+        return self._model.num_weights
+
+    @property
+    def weight_samples(self) -> np.ndarray:
+        """Returns network parameters (weights)."""
+        return self._weight_samples
+
+    @weight_samples.setter
+    def weight_samples(self, weight_samples: Union[List[float], np.ndarray, int]) -> None:
+        """Sets network parameters (weights)."""
+        if isinstance(weight_samples, int):
+            # random sampling from uniform distribution
+            self._weight_samples = algorithm_globals.random.uniform(
+                0, 1, size=(weight_samples, self._model.num_weights)
+            )
+        else:
+            weight_samples = np.asarray(weight_samples)
+            if (
+                len(weight_samples.shape) > 2
+                or weight_samples.shape[1] is not self._model.num_weights
```

Comment:
`is not` - do we mean `!=` here

Reply:
Oh, good point, of course it must be `!=`.